### PR TITLE
Add phi_1_5 hugging face model

### PR DESCRIPTION
**Pull request by pranavsharma**
Add phi_1_5 hugging face model https://huggingface.co/microsoft/phi-1_5

### CODE REVIEW

**Comment by pranavsharma** [comment]
@xuzhao9 - can you please take a look? Also, I tried to install this locally but ran into issues. See https://github.com/pytorch/benchmark/issues/1983

File: torchbenchmark/models/phi_1_5/__init__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+class Model(HuggingFaceModel):
+    task = NLP.LANGUAGE_MODELING
+    # TODO: is this needed given that this is a pre-trained model?
+    # DEFAULT_TRAIN_BSIZE = 6
+    DEFAULT_EVAL_BSIZE = 1
+
+    def __init__(self, test, device, jit=False, batch_size=None, extra_args=[]):
+        super().__init__(name="phi_1_5", test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
+
```
Comment (xuzhao9) [inline torchbenchmark/models/phi_1_5/__init__.py:12]: If the train test is not implemented, please add the `train(self)` function which raises the `NotImplemented` exception.

File: torchbenchmark/models/phi_1_5/__init__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from torchbenchmark.tasks import NLP
+from torchbenchmark.util.framework.huggingface.model_factory import HuggingFaceModel
+
+class Model(HuggingFaceModel):
+    task = NLP.LANGUAGE_MODELING
+    # TODO: is this needed given that this is a pre-trained model?
+    # DEFAULT_TRAIN_BSIZE = 6
```
Comment (xuzhao9) [inline torchbenchmark/models/phi_1_5/__init__.py:7]: Train test can still be added on a pre-trained model. If you are adding a train test, `DEFAULT_TRAIN_BSIZE` must be defined.

**Comment by xuzhao9** [comment]
Left a few comments. The code looks fine as long as it passes the CI check.

As for the installation error, we don't have testing environment on Windows, so we can't debug or resolve it.
However, we are glad to accept a fixing PR on Windows, as long as it does not break the Linux CI.

File: torchbenchmark/models/phi_1_5/__init__.py (hunk 1 above)
Comment (pranavsharma) [reply torchbenchmark/models/phi_1_5/__init__.py:12]: Done

**Comment by pranavsharma** [comment]
> Left a few comments. The code looks fine as long as it passes the CI check.
> 
> As for the installation error, we don't have testing environment on Windows, so we can't debug or resolve it. However, we are glad to accept a fixing PR on Windows, as long as it does not break the Linux CI.

I tried on Linux and at least the initial setup works. But when I run the test it says 0 tests.
 
(python310) pranav@pranav-HP-Z440-Workstation:~/work_projects/pytorch_benchmark$ python run.py phi_1_5 -d cpu -t eval
Running eval method from phi_1_5 on cpu in eager mode with input batch size 1 and precision fp32.
CPU Wall Time per batch: 4442.182 milliseconds
CPU Peak Memory:                6.6406 GB

(python310) pranav@pranav-HP-Z440-Workstation:~/work_projects/pytorch_benchmark$ python test.py -k "model_phi_1_5_eval_cpu"
 
----------------------------------------------------------------------
Ran 0 tests in 0.000s
 
OK

**Comment by xuzhao9** [comment]
Could you please try the command `python test.py -k "test_phi_1_5_eval_cpu"` ?

**Comment by pranavsharma** [comment]
> Could you please try the command `python test.py -k "test_phi_1_5_eval_cpu"` ?

This worked.
(python310) pranav@pranav-HP-Z440-Workstation:~/work_projects/pytorch_benchmark$ python test.py -k "test_phi_1_5_eval_cpu"
.
----------------------------------------------------------------------
Ran 1 test in 27.886s

OK

**Comment by pranavsharma** [comment]
@xuzhao9 - how do we move forward?

**Comment by xuzhao9** [comment]
@pranavsharma The model fails the tests on A10G but passes the tests on A100. What we could do is to skip the unit test on A10G for this model by updating `metadata.yaml`, example: https://github.com/pytorch/benchmark/blob/0d7b6afc7357cbc0dcd8c56336d4cff503ab1893/torchbenchmark/models/llama_v2_7b_16h/metadata.yaml#L9

**Review by xuzhao9 — APPROVED** [review]
LGTM

**Pull request closed without merging**